web/src/components/auth/password/forgot/ForgotPasswordForm.tsx

```diff
@@ -0,0 +1,154 @@
+import Button from '@mui/material/Button';
+import FormGroup from '@mui/material/FormGroup';
+import FormHelperText from '@mui/material/FormHelperText';
+import Stack from '@mui/material/Stack';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+import { useRouter } from 'next/navigation';
+import { useContext } from 'react';
+import { useForm } from 'react-hook-form';
+
+export interface ForgotPasswordValues {
+    email: string;
+    username: string;
+}
+
+const DEFAULT_VALUES: ForgotPasswordValues = {
+    email: '',
+    username: '',
+};
+
+const useAlert = () => {
+    return (message: string, type: string) => {
+        console.log(message, type);
+    };
+};
+
+const ForgotPasswordForm = () => {
+    const router = useRouter();
+    const showAlert = useAlert();
+    const { forgotPassword } = useContext(AuthContext);
+    const { formState, register, handleSubmit } = useForm<ForgotPasswordValues>(
+        USE_FORM_CONFIG(DEFAULT_VALUES, forgotPasswordSchema),
+    );
+    const { errors, isDirty, isValid, isSubmitting } = formState;
+
+    const onSubmit = async (formData: ForgotPasswordValues) => {
+        try {
+            setItem('username', formData.username);
+            await forgotPassword(
+                formData.username,
+                formData.email,
+                redirectToForgotPasswordCode,
+            );
+        } catch (error: unknown) {
+            if (isAWSError(error, 'CodeDeliveryFailureException')) {
+                showAlert((error as Error).message, 'error');
+            } else if (isAWSError(error, 'InvalidParameterException')) {
+                showAlert((error as Error).message, 'error');
+            } else if (isAWSError(error, 'NotAuthorizedException')) {
+                showAlert(
+                    'You are not authorized to perform this action.',
+                    'error',
+                );
+            } else if (isAWSError(error, 'TooManyRequestsException')) {
+                showAlert(TOO_MANY_REQUESTS(), 'error');
+            } else if (isAWSError(error, 'UserNotFoundException')) {
+                showAlert('User not found.', 'error');
+                router.push('/auth/login');
+            } else if (error instanceof Error) {
+                console.error(error.message);
+                showAlert(error.message, 'error');
+            } else {
+                console.error(error);
+                showAlert(UNKNOWN(), 'error');
+            }
+            removeItem('username');
+        }
+    };
+
+    const redirectToForgotPasswordCode = () => {
+        router.push('/auth/password/forgot/code');
+    };
+
+    const onCancel = () => {
+        router.back();
+    };
+
+    return (
+        <Box className='h-full'>
+            <Typography variant='h4'>Forgot Password</Typography>
+            <Typography variant='body1'>
+                Please provide your email and username. We will send you an
```

Comment:
Change this to username and email, since that's the order of the TextFields in the form.